lib/gs-metered.c

```diff
@@ -132,6 +132,11 @@ gs_metered_block_on_download_scheduler (GVariant      *parameters,
                                         GCancellable  *cancellable,
                                         GError       **error)
 {
+	if (g_getenv ("GS_UNIT_TESTS_SKIP_MOGWAI") != NULL) {
+		g_debug ("%s: Allowed to download (Skipping check in unit tests)", G_STRFUNC);
+		return TRUE;
+	}
```

Comment:
I’m not particularly enthusiastic about this or the parental controls environment variable. They mean we’re not actually testing the code we’re shipping, we’re testing some other code. Particularly with Mogwai and parental controls, this could result in the tests making it look like app installation works, but on actual systems it’s always disallowed through some interaction with Mogwai or parental controls.

The way to fix both of them is to run the gnome-software unit tests in a private D-Bus session bus, where the various daemons that it uses have been started by the test framework and configured to use that bus (setting `DBUS_SESSION_BUS_ADDRESS` in their environment). A custom D-Bus service directory might be needed, with a `.service` file for each service gnome-software uses, to allow service activation during the tests. Then `mogwai-scheduled` can be run as one of those services, and `accounts-daemon` as another, and that should get metered data and parental controls checks working.

A second phase of work on that would probably swap out `mogwai-scheduled` and `accounts-daemon` for mock implementations of their D-Bus interfaces, provided by the test harness. That would allow the results returned by them to be controlled by the tests. But I guess that for the first pass, just using the system-installed daemons and assuming that the configuration they load from the system is appropriate, will work.

If you don’t have time to do the first pass on that then I guess these environment variables will have to do. In that case I think they should definitely get a big `FIXME` comment each, pointing to tasks like https://phabricator.endlessm.com/T25340

Reply:
Completely agree that would be better but I don't think there's time for it right now. I'll add FIXME comments.

Reply:
Done